pgdog/src/frontend/client/query_engine/query.rs

```diff
@@ -264,9 +264,8 @@ impl QueryEngine {
     async fn transaction_error_check(
         &mut self,
         context: &mut QueryEngineContext<'_>,
-        route: &Route,
     ) -> Result<bool, Error> {
-        if context.in_error() && !context.rollback && route.is_cross_shard() {
```

Comment:
I don't remember why the cross-shard check is here. But I think this code is tested, so should be good.